test/jest/persistence.products.spec.ts

```diff
@@ -0,0 +1,34 @@
+import {
+    mParticle,
+    MPConfig,
+    testMPID,
+  } from '../src/config/constants';
+  import { Product } from '@mparticle/web-sdk';
+  
+  describe('Products Persistence', () => {
+      let mpInstance = mParticle.getInstance();
+      beforeEach(() => {
+          mParticle._resetForTests(MPConfig);
+          mpInstance._Persistence.resetPersistence();
+      });
+  
+      it('should save products to localStorage when noTargeting is false by default', () => {
+          const product = { Name: 'iphone', Sku: 'iphonesku', Price: 599, Quantity: 1 };
+          mpInstance._Persistence.setCartProducts({ [testMPID]: { cp: [product] } } as unknown as Product[]);
+          expect(mpInstance._Persistence.getCartProducts(testMPID).length).toBe(1);
```

Comment:
I think instead of trying to coerce the Type, let's see if we can use a `Partial` so we can maintain type safety.

```suggestion
          const product: Partial<Product> = { Name: 'iphone', Sku: 'iphonesku', Price: 599, Quantity: 1 };
          mpInstance._Persistence.setCartProducts({ [testMPID]: { cp: [product] } });
          expect(mpInstance._Persistence.getCartProducts(testMPID).length).toBe(1);
```

Reply:
@alexs-mparticle I’ve incorporated `Partial<Product>` for the item, but I’m still hitting the TypeScript error since `setCartProducts` is typed as `Product[]` while the runtime path expects a per‑MPID map. That’s why I had to coerce the call to match the actual shape earlier as well. If there’s a better approach or a preferred workaround to avoid the cast while keeping the test on the real write path, please advise and I’ll update.